src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -270,30 +306,83 @@ public Page<FeederResultEntity> findFeederResultsPage(ShortCircuitAnalysisResult
+    private void appendLimitViolationsAndFeederResults(Page<FaultResultEntity> pagedFaultResults,
+                                                       Sort.Order secondarySort,
+                                                       List<ResourceFilter> resourceFilters) {
         // using the Hibernate First-Level Cache or Persistence Context
         // cf.https://vladmihalcea.com/spring-data-jpa-multiplebagfetchexception/
         if (!pagedFaultResults.isEmpty()) {
             List<UUID> faultResultsUuids = pagedFaultResults.stream()
                     .map(FaultResultEntity::getFaultResultUuid)
-                    .collect(Collectors.toList());
+                    .toList();
+
             faultResultRepository.findAllWithLimitViolationsByFaultResultUuidIn(faultResultsUuids);
+
             faultResultRepository.findAllWithFeederResultsByFaultResultUuidIn(faultResultsUuids);
+            sortFeeders(pagedFaultResults, secondarySort);
+            filterFeeders(pagedFaultResults, resourceFilters);
+        }
+    }
+
+    private void filterFeeders(Page<FaultResultEntity> pagedFaultResults, List<ResourceFilter> resourceFilters) {
+        // feeders may only be filtered through connectableId
+        Optional<ResourceFilter> connectableIdFilter = resourceFilters.stream()
+                .filter(filter -> FeederResultEntity.Fields.connectableId.equals(filter.column()))
+                .findFirst();
+        connectableIdFilter.ifPresent(resourceFilter -> pagedFaultResults
+                .map(faultRes -> {
+                    List<FeederResultEntity> feeders = faultRes.getFeederResults()
+                            .stream()
+                            .filter(feeder -> feeder.match(resourceFilter)
+                            ).toList();
+                    faultRes.setFeederResults(feeders);
+                    return faultRes;
+                }));
+    }
+
+    private void sortFeeders(Page<FaultResultEntity> pagedFaultResults, Sort.Order secondarySort) {
+        // feeders may only be sorted by connectableId
+        if (secondarySort != null && FeederResultEntity.Fields.connectableId.equals(secondarySort.getProperty())) {
+            pagedFaultResults.map(res -> {
+                res.getFeederResults().sort(
+                    secondarySort.isAscending() ?
+                        Comparator.comparing(FeederResultEntity::getConnectableId) :
+                        Comparator.comparing(FeederResultEntity::getConnectableId).reversed()
+                );
+                return res;
+            });
+        } else {
+            // otherwise by default feederResults (within each individual faultResult) are sorted by 'current' in descending order :
+            pagedFaultResults.map(res -> {
+                res.getFeederResults().sort(
+                        Comparator.comparingDouble(FeederResultEntity::getCurrent).reversed());
+                return res;
+            });
```

Comment:
this is a bit weird, "current,desc" is default sort, but you can't set it yourself ? Or even set "current.asc" ?

Reply:
This is just a default sort. 

You can't set it yourself because this is a column which mixes parent and children data. Therefore when the user click on it a standard parent sort is used. This children sort is just there to give a basic standard order to the children columns and avoid to make them look "random".